crates/sound/src/audio_backends/pipewire.rs

```diff
@@ -341,15 +339,18 @@ impl AudioBackend for PwBackend {
                                     ptr::write_bytes(p.as_mut_ptr(), 0, n_bytes);
                                 }
                             } else {
-                                // consume() always reads (buffer.data_descriptor.len() -
-                                // buffer.pos) bytes
-                                buffer.consume(p).expect("failed to read buffer from guest");
-
-                                start += n_bytes;
-
-                                buffer.pos = start;
-
-                                if start >= buffer.data_descriptor.len() as usize {
+                                // consume() always tries to read p.len() unless error
+                                let n_bytes = match buffer.consume(p) {
+                                    Ok(n_bytes) => n_bytes as usize,
+                                    Err(_) => {
+                                        log::error!("Failed to read from guest memory");
+                                        0
```

Comment:
@dorindabassey do you think it is better to set `n_bytes` to `0` or just return if an error occurs?